subcommands/keys/ca_revoke_device_ca.go

```diff
@@ -0,0 +1,83 @@
+package keys
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/foundriesio/fioctl/client"
+	"github.com/foundriesio/fioctl/subcommands"
+	"github.com/foundriesio/fioctl/x509"
+)
+
+func init() {
+	cmd := &cobra.Command{
+		Use:   "revoke-device-ca <PKI Directory>",
+		Short: "Revoke device CA from the list of CAs allowed to issue device client certificates",
+		Run:   doRevokeDeviceCa,
+		Args:  cobra.ExactArgs(1),
+		Long: `Revoke device CA from the list of CAs allowed to issue device client certificates.
+
+When the online or local device CA is revoked:
+- It is no longer possible to register new devices with client certificates issued by this device CA.
+- Existing devices with client certificates issued by this device CA can no longer connect to your FoundriesFactory.
+
+You may later re-add a revoked device CA using "keys ca update", if you still keep its certificate stored somewhere.
+Once you do it, devices with client certificates issued by this device CA may connect to your FoundriesFactory again.`,
+	}
+	caCmd.AddCommand(cmd)
+	cmd.Flags().BoolP("dry-run", "", false,
+		"Do not revoke the certificate, but instead show a generated CRL that will be uploaded to the server.")
+	cmd.Flags().BoolP("pretty", "", false,
+		"Can be used with dry-run to show the generated CRL in a pretty format.")
+	cmd.Flags().StringArrayP("ca-file", "", nil,
+		"A file name of the device CA to revoke. Can be used several times to revoke several device CAs")
+	cmd.Flags().StringArrayP("ca-serial", "", nil,
+		"A serial number (base 10) of the device CA to revoke. Can be used several times to revoke several device CAs")
+	_ = cmd.MarkFlagFilename("ca-file")
+	// HSM variables defined in ca_create.go
+	cmd.Flags().StringVarP(&hsmModule, "hsm-module", "", "", "Load a root CA key from a PKCS#11 compatible HSM using this module")
+	cmd.Flags().StringVarP(&hsmPin, "hsm-pin", "", "", "The PKCS#11 PIN to log into the HSM")
+	cmd.Flags().StringVarP(&hsmTokenLabel, "hsm-token-label", "", "", "The label of the HSM token containing the root CA key")
+}
```

Comment:
Probably should add an `Examples` section here also. I saw that ca-serial string array was thinking "is that --ca-serial serial1,serial2 or --ca-serial serial1 --ca-serial serial2"

Reply:
I will try to add one; maybe also extract a part of the description into examples.

Reply:
https://github.com/foundriesio/fioctl/pull/324/commits/3cb1d444187b04a2464ea53b747b8fb651a26979 and https://github.com/foundriesio/fioctl/pull/324/commits/0467b2562c79ada90ea27f704eed5401cc5b3003